### PR TITLE
Added supression support for pet/mount modals and added mount modal

### LINKED ISSUE #8671: Mount Growing modal should use mount name, not pet name
[//]: # (Before logging this issue, please post to the Report a Bug guild from the Habitica website's Help menu. Most bugs can be handled quickly there. If a GitHub issue is needed, you will be advised of that by a moderator or staff member -- a player with a dark blue or purple name. It is recommended that you don't create a new issue unless advised to.)

[//]: # (Bugs in the mobile apps can also be reported there.)

[//]: # (If you have a feature request, use "Help > Request a Feature", not GitHub or the Report a Bug guild.)

[//]: # (For more guidelines see https://github.com/HabitRPG/habitrpg/issues/2760)

[//]: # (Fill out relevant information - UUID is found in Settings -> API)
### Description
[//]: # (Describe bug in detail here. Include screenshots if helpful.)
When you raise a pet into a mount, you get (assuming you haven't disabled it) a modal that says "You grew a [COLOR] [PETNAME]!" However, some mounts use a different term for the grown-up version than the pet version: Tiger vs. Tiger Cub, for instance. Given it's displaying the new form, it might make more sense for the modal to use the mount name than the pet name.

### LINKED ISSUE #9214: Popup suppression on pet hatching and mount raising is broken
[//]: # (Before logging this issue, please post to the Report a Bug guild from the Habitica website's Help menu. Most bugs can be handled quickly there. If a GitHub issue is needed, you will be advised of that by a moderator or staff member -- a player with a dark blue or purple name. It is recommended that you don't create a new issue unless advised to.)

[//]: # (Bugs in the mobile apps can also be reported there.)

[//]: # (If you have a feature request, use "Help > Request a Feature", not GitHub or the Report a Bug guild.)

[//]: # (For more guidelines see https://github.com/HabitRPG/habitica/issues/2760)

[//]: # (Fill out relevant information - UUID is found in Settings -> API)
### Description
[//]: # (Describe bug in detail here. Include screenshots if helpful.)

The user settings page includes checkboxes for "don't show popup when hatching a pet" and "don't show popup when raising a pet into a mount". However, these don't seem to have any effect on display of modals when performing these actions. Pet hatching always shows a modal, and mount raising never does. The modals should appear or not per the respective checkboxes.

### PR DESCRIPTION
**Pull request by TheHollidayInn**
Fixes #9214 
Fixes #8671

### CODE REVIEW

File: website/client/components/inventory/items/index.vue [hunk 1]
```diff
@@ -354,7 +354,9 @@ export default {
     },
     hatchPet (potion, egg) {
       this.$store.dispatch('common:hatch', {egg: egg.key, hatchingPotion: potion.key});
-      this.$root.$emit('hatchedPet::open', createAnimal(egg, potion, 'pet', this.content, this.user.items));
+      const newPet = createAnimal(egg, potion, 'pet', this.content, this.user.items);
```
Comment (negue) [inline website/client/components/inventory/items/index.vue]: I think this could be moved after `if`?

**Comment by negue** [comment]
- [ ] I selected `Don't show popup when hatching a pet` but I still get a `You hatched a new pet!` modal

- [x] the `Don't show popup when raising a pet into a mount` works

**Comment by TheHollidayInn** [comment]
Can you double check that the setting saved, I'm not seeing the modal after hatching.

**Comment by negue** [comment]
Now it works fine, the last time I checked it I also refreshed the session. Anyway now its working as expected :)

**Comment by negue** [comment]
Wait! I retested the pet to mount modal not the hatching pet one.


The pet one still opens even if the setting is present, I click in the stable on a pet to hatch it and the modal is still there

**Comment by TheHollidayInn** [comment]
Oh oops. Pet hatching. I was doing raising. Let me update that.

**Comment by negue** [comment]
the hatched pet modal is still there

**Comment by TheHollidayInn** [comment]
Can you screenshot your settings

**Comment by TheHollidayInn** [comment]
https://github.com/HabitRPG/habitica/pull/9882/files#diff-dd6f127b1cc799f848fcf1e6bb5f043eR359

**Comment by negue** [comment]
modal_setting

I get the "you hatched a pet" on `inventory/items` and `inventory/stable` too

**Comment by TheHollidayInn** [comment]
What steps are you taking to make them appear on the stable?

**Comment by negue** [comment]
`inventory/items`:
- drag egg on potion
- drag potion on egg

`inventory/stable`:
- click on the "hatchable pet paw" (black)
hatchable_pet
- then the modal opens to hatch the pet and => `hatch!`

**Comment by TheHollidayInn** [comment]
I didn't even know about the stable hatchable pet paw.

**Comment by TheHollidayInn** [comment]
I still can't replicate on the inventory items page. Can you check to ensure the setting is getting saved to the DB?

**Comment by SabreCat** [comment]
Two things:

* I think we should at least show a growl-style notification in the upper right, when hatching a pet with suppression on. Right now it's inconsistent with mount raising, and the lack of feedback makes it seem like something's gone wrong.
* I can't hatch pets from the Stable at all, with suppression on. Clicking the black paw has no effect.

**Comment by TheHollidayInn** [comment]
Can you hatch with suppression off by clicking the pets from the stable?

**Comment by TheHollidayInn** [comment]
Also, does the suppression work for hatching and raising in the normal ways though?

**Comment by SabreCat** [comment]
> Can you hatch with suppression off by clicking the pets from the stable?

Yes, once I uncheck the suppression option, I can hatch again.

> Also, does the suppression work for hatching and raising in the normal ways though?

Yup! Though for some of us, Stable hatching is the normal way, heh.

**Comment by TheHollidayInn** [comment]
Okay, updated! Thank you :D

**Comment by SabreCat** [comment]
Almost there! Now, all is working with suppression on. Two things with suppression off:

* "You grew your Fox-Shade!" in mount raising modal--shouldn't it be You grew your Shade Fox"? (If that's a preexisting thing, nvm)
* Hatching from Stable gives no modal even with suppression off.

**Comment by TheHollidayInn** [comment]
OKay, so I see there are actually two modals for hatching.

So, I am now adding the confirm modal back. The success modal for hatching doesn't currently exist on prod, so that's out of scope of this. 

Pushing fixes soon.

**Comment by TheHollidayInn** [comment]
Looks like the mount modal issue was here: #8671

**Comment by SabreCat** [comment]
Yeah, I think I commented out the success modal to work around the issue where you'd get an extra bogus blank hatching modal every time you hatched something. Nobody has complained, so maybe it'll just stay that way!

As for the "You raised a", not quite--that issue describes how you'd see "You raised a Golden Caterpillar!" instead of "You raised a Golden Butterfly!" I don't remember it saying "You raised a Caterpillar-Golden!"

I will check out the fixes momentarily!

**Comment by TheHollidayInn** [comment]
It should be fixed now either way. I used the mounts name, but maybe there is an issue with common.

**Comment by SabreCat** [comment]
We're back to having popups for Stable hatching with suppression on ;_;

**Comment by TheHollidayInn** [comment]
Those are confirmation popups which is different.

**Pull request merged**